Tribler/community/trustchain/community.py

```diff
@@ -224,3 +249,138 @@ def unload_community(self):
         yield super(TrustChainCommunity, self).unload_community()
         # Close the persistence layer
         self.persistence.close()
+
+    def set_live_edge_callback(self, func):
+        """
+        Set the callback function for live edge updates.
+        Passed arguments are:
+          live_edge_id, [candidates]
+        """
+        self._live_edge_cb = func
+
+    def reset_live_edges(self):
+        """
+        Reset the live edges counter and current live edge.
+        """
+        self._live_edge = []
+        self._live_edge_next = None
+        self._live_edge_id = 0
+
+    def set_live_edges_enabled(self, value):
+        """
+        Enable or disable live edges.
+
+        :param value: whether or not to enable live edges
+        :type value: boolean
+        """
+        if value and not self._live_edges_enabled:
+            # Make sure we don't inherit old edge data after a reset
+            self.reset_live_edges()
+        self._live_edges_enabled = value
+
+    def get_trust(self, member):
+        """
+        Get the trust for another member.
+        Currently this is just the amount of MBs downloaded from them.
+
+        :param member: the member we interacted with
+        :type member: dispersy.member.Member
+        :return: the trust value for this member
+        :rtype: int
+        """
+        block = self.persistence.get_latest(member.public_key)
+        if block:
+            return block.sequence_number
+        else:
+            return 1
+
+    def dispersy_get_introduce_candidate(self, exclude_candidate=None):
+        """
+        Choose a trusted candidate to introduce to someone else.
+        The more trust you have for someone, the higher the chance is to forward them.
+        """
+        if not self._live_edges_enabled:
+            return super(TrustChainCommunity, self).dispersy_get_introduce_candidate(exclude_candidate)
+
+        eligible = [candidate for candidate in self._candidates.itervalues()
+                    if candidate.get_member() and candidate != exclude_candidate]
+
+        if not eligible:
+            # If we have no trusted candidates, bootstrap this process.
+            return super(TrustChainCommunity, self).dispersy_get_introduce_candidate(exclude_candidate)
+
+        total_trust = sum([self.get_trust(candidate.get_member()) for candidate in eligible])
+
+        random_trust_i = randint(0, total_trust - 1)
```

Comment:
Please add some comment above this piece of code to explain how the candidate selection works.